spec.html

```diff
@@ -39012,7 +38971,6 @@ <h1>Properties of the GeneratorFunction Prototype Object</h1>
         <li>is an ordinary object.</li>
         <li>is not a function object and does not have an [[ECMAScriptCode]] internal slot or any other of the internal slots listed in <emu-xref href="#table-27"></emu-xref> or <emu-xref href="#table-56"></emu-xref>.</li>
         <li>is the value of the *"prototype"* property of %GeneratorFunction%.</li>
-        <li>is the intrinsic object <dfn>%Generator%</dfn> (see Figure 2).</li>
```

Comment:
`%GeneratorFunction.prototype%` only works because the `prototype` property is already defined; in other words, the first of those two is the one that's actually required. The second one isn't necessary, I suppose, but is nice for clarity.

Reply:
If the spec says "X is %Foo.bar%", then by definition, that means that if I were to access the 'bar' property of '%Foo% prior to any ECMAScript code being evaluated, I would get X. To me, this seems equivalent to saying that X is the initial value of the 'bar' property of %Foo%. You're saying these aren't equivalent?

Reply:
@jmdyck i'm saying that relying on that kind of inference isn't something I'd prefer; `%Foo.bar%` works *because* it's the initial value of `bar` on `Foo`, which imo should have to be defined somewhere.

Reply:
Isn't [25.2.2.2 Generator.prototype](https://tc39.es/ecma262/#sec-generatorfunction.prototype) enough of a definition?

Reply:
25.2.2.2 there doesn't actually have an explicit link that I can see to 25.2.3, "the GeneratorFunction prototype object"; 25.2.3 has "is the value of the "prototype" property of %GeneratorFunction%." which is the link.

Reply:
I'm confused, so I'll back up. Earlier, I asserted the equivalence of the two statements:
- X is %Foo.bar%
- X is the initial value of the `bar` property of %Foo%.

You didn't deny that they were equivalent, but you said that the first relies on a kind of inference (on the part of the reader, presumably) that you didn't prefer. I don't really understand that, but I can probably accept it as an editorial preference.

So, to clarify the specifics of this preference, 2 questions:

1) Would it make a difference for you if the statement were the other way round:
    ```%Foo.bar% is X``` ?

2) Do you expect to eventually remove all statements of the form ```The initial value of `Foo.prototype` is %Foo.prototype%.```, or do any of them serve a purpose?

Reply:
I'm saying that they're not equivalent unless the second one exists.

Reply:
To clarify, when you say "the second one exists", do you mean "somewhere in the spec establishes that %Foo% **has** a `bar` property"?

Reply:
Something that establishes that Foo has a bar property, and that the value of that property is "the object in question" (ie, "the Array prototype object", or whatever)